### PR TITLE
Add test to check empty mirrorset creation fails

### PR DESCRIPTION
**Pull request by mayankkapoor**

### CODE REVIEW

File: aasemble/django/apps/api/tests.py [hunk 1]
```diff
@@ -400,3 +400,20 @@ def test_github_push_event(self, poll_one):
...
         github_push_event("https://github.com/eric/project0")
         poll_one.delay.assert_called_with(1)
+
+
+class APIv1MirrorsetTests(APIv1Tests):
+    list_url = '/api/v1/mirror_sets/'
+
+    def test_create_mirrorset_empty_fails_400(self):
+        # TODO: The funny thing is, this scenario succeeds in creating mirrorset
+        # on the prod server returning a status code of 201 CREATED. Check why.
```
Comment (sorenisanerd) [inline aasemble/django/apps/api/tests.py:410]: Wow. Yeah, this means we can't really trust these test results. That's very interesting. Let's get to the bottom of this ASAP.

Comment (sorenisanerd) [inline aasemble/django/apps/api/tests.py:410]: It's not just production. If I run a local dev server (`python manage.py runserver`) I get the same result (i.e. no error, just an empty mirror set).

Comment (sorenisanerd) [inline aasemble/django/apps/api/tests.py:410]: Wow. It took a while to work this out.

The difference is: `self.client.post(url, {}, format='json')` actually sends an empty json dictionary.

Sending an empty request using curl or httpie or whatever sends _nothing_. Nothing gets turned into a QueryDict. A QueryDict is handled slightly differently than a regular dictionary.

Comment (sorenisanerd) [inline aasemble/django/apps/api/tests.py:410]: https://github.com/tomchristie/django-rest-framework/issues/3647
